Support `cpu.family` hardware requirement for `mrack`

skycastlelily opened the pull request:
Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] extend the test coverage
* [x] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [x] mention the version
* [ ] include a release note

happz commented on tmt/steps/provision/mrack.py:
```diff
@@ -229,6 +229,18 @@ def _transform_beaker_pool(
 
 
+def _transform_cpu_family(
+        constraint: tmt.hardware.NumberConstraint,
```
Should be `IntegerConstraint`, CPU family will never be a floating point number.

skycastlelily commented:
Yeah, this one is not based on cpu-frequency,so it didn't know about
IntegerConstraint,now
rebased, and updated^^

On Sat, Nov 2, 2024 at 5:19 PM Miloš Prchlík ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tmt/steps/provision/mrack.py
> <https://github.com/teemtee/tmt/pull/3302#discussion_r1826532548>:
>
> > @@ -229,6 +229,18 @@ def _transform_beaker_pool(
>          actual_value)
>
>
> +def _transform_cpu_family(
> +        constraint: tmt.hardware.NumberConstraint,
>
> Should be IntegerConstraint, CPU family will never be a floating point
> number.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/teemtee/tmt/pull/3302#pullrequestreview-2411221185>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKFR23B35MHXKSKMCSPHCULZ6SKJJAVCNFSM6AAAAABQFBY5VCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDIMJRGIZDCMJYGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

skycastlelily commented:
Yeah, this one is not based on cpu-frequency,so it didn't know about IntegerConstraint,now rebased, and updated^^

happz commented:
@skycastlelily I think the missing piece is a note in release notes & an update for HW requirements stating `cpu.family` is now supported by `beaker` plugin.

thrix reviewed (approved):
LGTM with the requested docs change

psss commented:
Failing `el9` job is irrelevant to this, merging.
